lib/filter_param.cpp

```diff
@@ -700,3 +700,97 @@ vector<double> FilterParam::init_stable_coef(const double a0, const double a) co
 	return coef;
 }
 
+/* # フィルタ構造体
+ *   振幅特性図の描画
+ * 	 leftとrightで描画範囲の指定[0:0.5]  
+ *
+ * # 引数
+ * vector<double> &coef : 係数列
+ * string &filename : 出力するファイル名
+ * double left : 帯域の左端正規化周波数 [0:0.5)
+ * double right : 帯域の右端正規化周波数 (0:0.5]
+ * 
+ */
+void FilterParam::gprint_amp
+(const vector<double> &coef, const string &filename, const double left, const double right) const
+{
+	int i = 0;
+
+	auto freq_res = FilterParam::freq_res(coef);
+	vector<BandParam> range{FilterParam::gen_bands(FilterType::LPF, left, right)};
+
+	//gnuplotで出力
+    FILE *gp = popen("gnuplot -persist", "w");
+    fprintf(gp, "set terminal pngcairo\n");
+	fprintf(gp, "set output 'Amplitude.png'\n");
+    fprintf(gp, "set xlabel 'Nomalized angular frequeny'\n"); //正規化角周波数
+    fprintf(gp, "set ylabel 'Amplitude'\n");
+	fprintf(gp, "set key    font 'Times New Roman,10'\n");
+	fprintf(gp, "set xlabel font 'Times New Roman,12'\n");
+	fprintf(gp, "set ylabel font 'Times New Roman,12'\n");
+	fprintf(gp, "set tics   font 'Times New Roman,10'\n");
+	fprintf(gp, "set xtics 0, 0.1*pi, pi\n");
+	fprintf(gp, "set yrange [0:1.2]\n");
+	fprintf(gp, "set format x '%.1Pπ'\n");
+    fprintf(gp, "plot '-' with lines title \"\n");
+
+   	for(auto band_res :freq_res)
+	{
+		for(auto res :band_res)
+		{
+            i += 1;
+
+			if(left*2*M_PI <= i*(M_PI/250) && i*(M_PI/250) <= right*2*M_PI)
+			{
+				printf("%lf %f\n", i*(M_PI/250), abs(res));
+				fprintf(gp,"%lf %f\n", i*(M_PI/250), abs(res));
+			}
+		}
```

Comment:
```suggestion
		for(unsigned int i = 0 ; i < band_res.size() ; ++i)
		{
			if(left*2*M_PI <= i*(M_PI/250) && i*(M_PI/250) <= right*2*M_PI)
			{
				printf("%lf %f\n", i*(M_PI/250), abs(res));
				fprintf(gp,"%lf %f\n", i*(M_PI/250), abs(res));
			}
		}
```
インデックス使いたいなら、インデックスで回そう
あと条件判定は関数冒頭で1回だけ行う